### DISCUSSION TITLE
Store OAuth settings separately

### DISCUSSION THREAD

**Discussion by pedorich-n** [immich-app/immich#9223] [Feature Request]
Hi!
I would like to have my Immich config backed up in a git repository, but because I use OAuth my config file contains secret values that I'd like to keep away from git. It would be nice to have the OAuth section of the config stored somewhere else, perhaps in the DB, or a separate file.
I realize that it doesn't make much sense from the UI and the backend POV, but having those secret values in plain json gives me shivers. 

Is there something that can be done?

**Comment by bo0tzz** [comment]
You could have a script that merges the config json from different sources before deploying Immich. Another option is to use something like https://github.com/getsops/sops to encrypt the file in git. There's plenty of good tools for doing stuff like this already, so I don't think we'll include it in Immich.

**Reply by pedorich-n** [reply]
Yeah, I am already using [age](https://github.com/FiloSottile/age), but I was hoping to avoid encrypting the entire config since it doesn't have to be. 
I guess the only viable option for me is templating/merging the files.
Thanks!
